lib/pluginUtils.js

```diff
@@ -84,6 +86,16 @@ function _checkDecorators (instance, decorators) {
   })
 }
 
+function checkVersion (fn) {
+  const meta = getMeta(fn)
+  if (!meta) return
+
+  const requiredVersion = meta.fastify
+  if (!requiredVersion) return
+
+  if (semver.lte(this.version, requiredVersion)) throw new FST_ERR_PLUGIN_VERSION_MISMATCH(meta.name, requiredVersion, this.version)
```

Comment:
In our plugins we have patterns like `'>=3.x'`

In this case `semver.lte('3.3.1', '>=3.x')` it will throw or Am I missing something?

https://github.com/fastify/fastify-oauth2/blob/master/index.js#L144

Reply:
Indeed will have to use validerange of semver. Will fix and add tests for this. Good catch